feat(replay): Add example for `beforeAddRecordingEvent` (#7547)
getsentry/sentry-docs: src/platforms/javascript/common/session-replay/privacy.mdx

});
```

### Example: Capturing 500 Status Codes Only

> Here is an example that will only capture fetch requests that return a 500 status code. Non-fetch requests will continue to be captured as normal.

Reviewer: ```suggestion
Here's an example showing how to only capture fetch requests that return a 500 status code. (Non-fetch requests would continue to be captured normally.)
```

```javascript
new Sentry.Replay({
  beforeAddRecordingEvent: (event) => {
    // Do not capture fetch/xhr requests, unless the response code is 500
    if (
      event.data.tag === "performanceSpan" &&
      (event.data.payload.op === "resource.fetch" ||
        event.data.payload.op === "resource.xhr") &&
      event.data.payload.data.statusCode !== 500
    ) {
      return null;
    }

    return event;
  },
});
```

> Note that we also have [server-side PII scrubbing](/product/data-management-settings/scrubbing/server-side-scrubbing/) for this data. It looks for certain patterns such as American social security numbers, credit cards, and private keys.

Reviewer: ```suggestion
We also have [server-side PII scrubbing](/product/data-management-settings/scrubbing/server-side-scrubbing/) for this data. It looks for certain patterns such as American social security numbers, credit cards, and private keys.
```
Would be good to get rid of "Note" here since there's another sentence that starts with "Note" right below :)

});
```

### Example: Capturing 500 Status Codes Only

Here is an example that will only capture fetch requests that return a 500 status code. Non-fetch requests will continue to be captured as normal.

```javascript
new Sentry.Replay({
  beforeAddRecordingEvent: (event) => {

>     // Capture non-fetch requests as normal
>     if (
>       event.data.tag !== "performanceSpan" ||
>       event.data.payload.op !== "resource.fetch"
>     ) {
>       return event;
>     }
> 
>     // Only capture requests that result in a 500 status code
>     if (event.data.payload.data.statusCode !== 500) {
>       return null;
>     }
> 
>     return event;

Reviewer 2: IMHO The negation in the first stanza, and duplicate `return event` statements, make this harder to understand if someone's not as familiar with the types.

what about this:

```suggestion
    // Do not capture fetch/xhr requests, unless the response code is 500
    if (
      event.data.tag === "performanceSpan" &&
      (event.data.payload.op === "resource.fetch" || event.data.payload.op === 'resource.xhr') &&
      event.data.payload.data.statusCode !== 500
    ) {
      return null;
    }
    
    return event;
```

Author: Yeah this makes more sense.
